### PR TITLE
Explicitly list types in iszero test

### PR DESCRIPTION
**Pull request by ararslan**
This addresses @tkelman's comment in https://github.com/JuliaLang/julia/pull/19950#discussion_r96108525. I've restructured the `iszero` tests such that the types are explicitly listed out. I've also consolidated more of the other tests into the loop.

### CODE REVIEW

**Review by Sacha0 — APPROVED** [review]

File: test/numbers.jl [hunk 1]
```diff
@@ -2867,14 +2867,14 @@ end
 
 @testset "iszero" begin
     # Numeric scalars
-    for T in Iterators.flatten(subtypes.([AbstractFloat, Signed, Unsigned]))
+    for T in [Float16, Float32, Float64, BigFloat,
+              Int8, Int16, Int32, Int64, Int128, BigInt,
+              UInt8, UInt16, UInt32, UInt64, UInt128,
+              Rational]
```
Comment (stevengj) [inline test/numbers.jl]: This makes me realize that we could use `iszero(x::Rational) = iszero(x.num)`, which saves a comparison compared to the fallback method that does `== 0`

File: test/numbers.jl [hunk 2]
```diff
@@ -2867,14 +2867,14 @@ end
...
         @test iszero(T(0))
         @test iszero(Complex{T}(0))
```
Comment (stevengj) [inline test/numbers.jl]: For if `T<:AbstractFloat`, we should also check `iszero(T(-0.0))`

File: test/numbers.jl (hunk 1 above)
Comment (stevengj) [inline test/numbers.jl]: Rather than including `Rational` in this list, I would check `iszero(Rational{T}(0))` if `T <: Integer`.

**Comment by ararslan** [comment]
Great ideas, @stevengj! I've incorporated the changes mentioned in your comments. Thanks!

**Comment by ararslan** [comment]
How are things looking now?

**Pull request merged**